binary_transparency/firmware/internal/ftmap/mapdb.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ftmap
+
+import (
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/google/trillian/experimental/batchmap"
+)
+
+// NoRevisionsFound is returned when the DB appears valid but has no revisions in it.
+type NoRevisionsFound = error
+
+// MapDB provides read/write access to the generated Map tiles.
+type MapDB struct {
+	db *sql.DB
+}
+
+// NewMapDB creates a MapDB using a file at the given location.
```

Comment:
/using/in/ ?

Reply:
I'll leave as "using". It needs to work for both writes and reads. I think "using a file" makes sense. This will need to be replaced with a client/server compatible version to really exercise Dataflow down the line anyway.